public/src/modules/Tunnel.js

```diff
@@ -75,38 +80,21 @@ export default class Tunnel extends CircuitElement {
         // rect2(ctx, -120 + xRotate + this.xSize, -20 + yRotate, 120 - this.xSize, 40, xx, yy, "RIGHT");
     }
 
-    /**
-     * @memberof Tunnel
-     * function to set tunnel value
-     * @param {number} val - tunnel value
-     */
-    setTunnelValue(val) {
-        this.inp1.value = val;
-        for (let i = 0; i < this.inp1.connections.length; i++) {
-            if (this.inp1.connections[i].value !== val) {
-                this.inp1.connections[i].value = val;
-                simulationArea.simulationQueue.add(this.inp1.connections[i]);
-            }
-        }
-    }
-
     /**
      * @memberof Tunnel
      * resolve output values based on inputData
      */
     resolve() {
-        for (
-            let i = 0;
-            i < this.scope.tunnelList[this.identifier].length;
-            i++
-        ) {
-            if (
-                this.scope.tunnelList[this.identifier][i].inp1.value !==
-                this.inp1.value
-            ) {
-                this.scope.tunnelList[this.identifier][i].setTunnelValue(
-                    this.inp1.value
-                );
+        // Check for bitwidth error since it bypasses node's resolve() function which usually checks bitwidths
+        for (const tunnel of this.scope.tunnelList[this.identifier]) {
+            if (tunnel.inp1.bitWidth !== this.inp1.bitWidth) {
+                this.inp1.highlighted = true;
+                tunnel.inp1.highlighted = true;
+                showError(`BitWidth Error: ${this.inp1.bitWidth} and ${tunnel.inp1.bitWidth}`);
+            }
+            if (tunnel.inp1.value !== this.inp1.value) {
+                tunnel.inp1.value = this.inp1.value;
+                simulationArea.simulationQueue.add(tunnel.inp1);
```

Comment:
This will actually be a N2 algorithm u no? Each tunnel checks if other tunnels are updated. This was how it was before also. Can we eliminate it somehow?

Reply:
We can make another boolean array that holds whether the tunnel was checked. Then if the element is checked we just do checked[i] = false and return instead of resolving

Reply:
@satu0king can you please review